src/accelerate/accelerator.py

```diff
@@ -1025,7 +1025,7 @@ def accumulate(self, *models):
         ...         optimizer.zero_grad()
         ```
         """
-        self._do_sync()
+        self._do_sync(force=self.gradient_state.plugin_kwargs.get("sync_each_batch", False))
```

Comment:
Similarly no need for a `get`

Reply:
@muellerzr the `get` is for the kwarg key `sync_each_batch`. Turns out that [KwargsHandler only returns a key-value if the value is non-default](https://github.com/huggingface/accelerate/blob/10882eeddd841b0a1a8064421816ee0daef99920/src/accelerate/utils/dataclasses.py#L57). Hence if `sync_each_batch` was set to `False`, then it will be missing in `plugin_kwargs`.

[See here for how `plugin_kwargs` are populated in `GradientState`](https://github.com/fabianlim/accelerate/blob/329327b0507686f8cf0c0748849cd4858aa5b37b/src/accelerate/state.py#L1034)